### DISCUSSION TITLE
useFieldArray doesn't behave as expected in a nested form

### DISCUSSION THREAD

**Discussion by fowku** [react-hook-form/react-hook-form#10241] [Q&A]
### Version Number

7.43.8

### Codesandbox/Expo snack

https://codesandbox.io/s/usefieldarray-conditional-forked-rrhxvo?file=/src/NestedForm.tsx

### Steps to reproduce

1. Go to [sandbox](https://codesandbox.io/s/usefieldarray-conditional-forked-rrhxvo?file=/src/NestedForm.tsx)
3. Click on minimum first input ![image](https://i.imgur.com/WCHesA6.png)
4. Change 10 to another number value
5. Press "x" on the right of the first minimum input
6. Default value is returned to the field
7. Press submit and see console.log. Field is undefined, but control shows default value ![image](https://i.imgur.com/8tG8qp9.png)

### Expected behaviour

Input should be cleared, disabled field value have to be recalculated.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fowku** [comment]
Actually, I missed this in docs. 
<img width="481" alt="image" src="https://user-images.githubusercontent.com/28464788/230401442-07382a6b-0201-4812-8f60-36822833918c.png">
But what should I do, if my UI lib uses undefined only? Is there any way to fix it in my app?

**Comment by devakrishna33** [comment]
@fowku Try using resetField from [https://react-hook-form.com/api/useform/resetfield/](https://react-hook-form.com/api/useform/resetfield/), it will fix your issue

**Comment by fowku** [comment]
@devakrishna33 unfortunately, it has same effect as setValue(*, undefined). I see that my example uses defaultValues wrong.

**Comment by devakrishna33** [comment]
@fowku I have updated your sandbox with resetField, please check this 
[https://codesandbox.io/s/usefieldarray-conditional-forked-ein5v7?file=/src/NestedForm.tsx](https://codesandbox.io/s/usefieldarray-conditional-forked-ein5v7?file=/src/NestedForm.tsx)

**Comment by fowku** [comment]
@devakrishna33 thank you! But as I can see, field is still being reset to the defaultValue, but I want field to be cleared.

**Comment by bluebill1049** [comment]
We do not support `undefined` value, this is a limitation due to inline `defaultValue` vs `defautlValues` from `useForm`, to support `undefined` would leak to a major breaking change.

**Comment by leapful** [comment]
@fowku 

Just put default value 0 on param of `onChange` callback to solve this issue since your NumberInput is always a number. In this case, undefined or 0 are the same for BE handling.

Working demo: https://codesandbox.io/s/usefieldarray-conditional-forked-hk4z82
